### ISSUE TITLE
API documentation plugin can’t see custom controller

### ISSUE THREAD

**Issue by violabg** [strapi/strapi#13403]
### Required System information

- Node.js version: 16
- NPM version:
- Strapi version: 4.1.11
- Database: postgres 
- Operating system: macOS

### Describe the bug
After generating API documentation with the relative plugin, I’m only able to see basic CRUD method and not the custom one I have created.

### Steps to reproduce the behavior

Extend and API route and add a new controller.
Install the API Documentation plugin
Generate the documentation

### Expected behavior
To see custom API methods

### Code snippets

If applicable, add code samples to help explain your problem.

**Comment by emiller12** [comment]
I think I ran into this one today as well. 

I have one controller in my application where I am overriding the default `findOne` method with my own implementation as follows:

```
const { createCoreController } = require('@strapi/strapi').factories;
module.exports = createCoreController('api::client.client', ({ strapi }) =>  ({
    async findOne(ctx) {
        const uuid = ctx.params.id;
        const response = await strapi.query('api::mytype.mytype').findOne({
            where: {
                customField: uuid
            },
            populate: true
        });
        return response;
    }
}));
```

This returns an object from the api like this:

```
{ data: { ...myFields } }
```

Where as the documentation that is generated by the plugin still assumes that the format is the default one nested under the attributes key:
```
{ data: { attributes: { ...myFields } } }
```

**Comment by MHZarei** [comment]
I have the same problem with custom routes and controller.
In this repository I created custom route:
https://github.com/MHZarei/strapi-api-documentation/blob/master/src/api/testapi/routes/testapi.js
```

module.exports = {
  routes: [
    {
     method: 'GET',
     path: '/testapi',
     handler: 'testapi.exampleAction',
     config: {
       policies: [],
       middlewares: [],
       "description": "a testemail api",
       "tag": {
         "plugin": "testapi",
         "name": "Testapi"
       }
     },
    },
  ],
};
```

[According to documentation ](https://github.com/strapi/strapi/tree/master/packages/plugins/documentation#how-does-it-generate-the-others-plugins-documentation-)add "description" to config section, but after generate documentation only show CRUD and user-permission methods.

Can you tell me how add custom routes to documentation?

**Comment by emilieringwald** [comment]
Hey @violabg ! Thanks for the ticket and sorry for the delay, we added it to our backlog and will review it asap.

**Comment by spyrocash** [comment]
> I have the same problem with custom routes and controller in strapi 4.3.3. In this repository I created custom route: https://github.com/MHZarei/strapi-api-documentation/blob/master/src/api/testapi/routes/testapi.js
> 
> ```
> 
> module.exports = {
>   routes: [
>     {
>      method: 'GET',
>      path: '/testapi',
>      handler: 'testapi.exampleAction',
>      config: {
>        policies: [],
>        middlewares: [],
>        "description": "a testemail api",
>        "tag": {
>          "plugin": "testapi",
>          "name": "Testapi"
>        }
>      },
>     },
>   ],
> };
> ```
> 
> [According to documentation ](https://github.com/strapi/strapi/tree/master/packages/plugins/documentation#how-does-it-generate-the-others-plugins-documentation-)add "description" to config section, but after generate documentation only show CRUD and user-permission methods.
> 
> Can you tell me how add custom routes to documentation?

same here.

**Comment by Darthnerda** [comment]
> I have the same problem with custom routes and controller in strapi 4.3.3. In this repository I created custom route: https://github.com/MHZarei/strapi-api-documentation/blob/master/src/api/testapi/routes/testapi.js
> 
> ```
> 
> module.exports = {
>   routes: [
>     {
>      method: 'GET',
>      path: '/testapi',
>      handler: 'testapi.exampleAction',
>      config: {
>        policies: [],
>        middlewares: [],
>        "description": "a testemail api",
>        "tag": {
>          "plugin": "testapi",
>          "name": "Testapi"
>        }
>      },
>     },
>   ],
> };
> ```
> 
> [According to documentation ](https://github.com/strapi/strapi/tree/master/packages/plugins/documentation#how-does-it-generate-the-others-plugins-documentation-)add "description" to config section, but after generate documentation only show CRUD and user-permission methods.
> 
> Can you tell me how add custom routes to documentation?

I am also having this issue. It might be that the route is unrelated to a content-type. When I changed my route to be an extension on the user-permissions plugin user content-type, I at least got the path showing up in the documentation (though the description field didn't do anything).

**Comment by duchmse23** [comment]
Hi, are there any news on the current issue? I also experiencing same problem when documentation plugin does not generate document for custom controllers.

**Comment by shfx** [comment]
Hi, my team is considering using this plugin but lack of custom controllers is kinda no-go for us. Is this something hard to fix/implement? Not sure how Swagger extracts types form other controllers. Is it reflecting on types?

**Comment by MoeedthePessimist** [comment]
Any solutions yet? Or is there any other way to override the documentation ourselves by preventing strapi from removing the custom documentation and replacing it with empty objects?

**Comment by markkaylor** [comment]
At the moment I don't have the bandwidth to work on this issue. I would be happy to review a PR if someone is interested in working on this.

@MoeedthePessimist it is possible to override the documentation, please see the docs: https://docs.strapi.io/dev-docs/plugins/documentation#overriding-the-generated-documentation and let me know if something is not clear.

**Comment by MoeedthePessimist** [comment]
@markkaylor Yes I tried the way in the given documentation. However I couldn't find a way to categorize it under an api. It always ends up showing the endpoint's override documentation under the **default** tab. 

Let me know if I am missing out on something. It will be very helpful.

**Comment by markkaylor** [comment]
@MoeedthePessimist I think you are most likely missing tags in your schema override https://swagger.io/docs/specification/grouping-operations-with-tags/. 

Your questions about the overrides is not related to this issue. Please ask them in the community discord or Strapi forum. If you think there is an issue with the overrides then please open a new issue. Thank you.

**Comment by rrubio** [comment]
Any updates on this? i'm having the same issue on strapi 4.13.6 (node v18.16.0).

**Comment by haotangio** [comment]
> Any updates on this? i'm having the same issue on strapi 4.13.6 (node v18.16.0). Mine is a custom api. Under the documentation for this api the documentation folder was generated with an empty object.
> 
> e.g.
> 
> myapi.json
> 
> ```
> {}
> ```

Same issue here.
Can someone check it. This is a big issue as it blocks the community from writing great plugin. The manual process of writing documentation is absolutely sick.

**Comment by ayhid** [comment]
Do you think creating annotations on custom routes could help generating documentation for them?

I am willing to work on this as I need it for a project , but TBH I don't know where to start.

Maybe an extra configuration key in routes definition?

**Comment by Shekhar-Zealous** [comment]
Strapi Version: 4.23.0

I am facing a issue in Plugin documentation after it's installation and only see REST APIs and some of my custom APIs from the controller in the Swagger documentation. 

It's strange that some are listed and some are not. The reason is unknown to me.

Is there any issue with the plugin documentation?

**Comment by Abdulrahman-AL-Sabagh** [comment]
It seems like someone found a solution here
https://forum.strapi.io/t/how-to-add-custom-routes-to-core-routes-in-strapi-4/14070/7
